creator/data_templates/mutations/template_version.py

```diff
@@ -0,0 +1,257 @@
+from pprint import pprint
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.db import transaction
+from django.conf import settings
+
+from creator.studies.models import Study
+from creator.data_templates.models import (
+    DataTemplate,
+    TemplateVersion,
+)
+from creator.data_templates.nodes.template_version import TemplateVersionNode
+
+
+def check_studies(study_node_ids):
+    """
+    Check if studies exist and return the Study objects if all studies exist
+
+    Helper function called in create/update template version mutations
+    """
+    # Convert graphql relay ids to primary keys
+    study_ids = []
+    for sid in study_node_ids:
+        _, study_id = from_global_id(sid)
+        study_ids.append(study_id)
+
+    # Check all studies exist
+    studies = Study.objects.filter(pk__in=study_ids).only("kf_id").all()
+    if len(studies) != len(study_ids):
+        raise GraphQLError(
+            "Failed to create/update template version because one or more "
+            "studies in the input do not exist. "
+        )
+    return studies
+
+
+class CreateTemplateVersionInput(graphene.InputObjectType):
+    """ Parameters used when creating a new template_version """
+
+    description = graphene.String(
+        required=True, description="The description of the template_version"
+    )
+    field_definitions = graphene.JSONString(
+        required=True,
+        description="The field definitions for this template"
+    )
+    data_template = graphene.ID(
+        required=True,
+        description="The data_template that this template_version belongs to"
+    )
+    studies = graphene.List(
+        graphene.ID,
+        description="The studies this template version should be assigned to"
+    )
+
+
+class UpdateTemplateVersionInput(graphene.InputObjectType):
+    """ Parameters used when updating an existing template_version """
+
+    description = graphene.String(
+        description="The description of the template_version"
+    )
+    field_definitions = graphene.JSONString(
+        description="The field definitions for this template"
+    )
+    studies = graphene.List(
+        graphene.ID,
+        description="The studies this template version should be assigned to"
+    )
+
+
+class CreateTemplateVersionMutation(graphene.Mutation):
+    """ Creates a new template_version """
+
+    class Arguments:
+        input = CreateTemplateVersionInput(
+            required=True, description="Attributes for the new template_version"
+        )
+
+    template_version = graphene.Field(TemplateVersionNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new template_version.
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.add_datatemplate"):
```

Comment:
I wasn't sure if template version should have their own permissions or just use the parent data template's permissions. I just did the latter for now bc its easy and could be changed later